### PR TITLE
separate feature detect from polyfill

### PR DESCRIPTION
**Pull request by JakeChampion**
With the two together, it is impossible to test whether the module
works in engines which partially support Array.of such as node v5.

With the two separated, we can test in all engines and we can more
easily use this polyfill inside of the [Financial Times's polyfill-service](https://github.com/Financial-Times/polyfill-service).

As this pull request contains a lot of whitespace changes, here is a link which should make the actual changes easier to view -- https://github.com/mathiasbynens/Array.of/pull/8/files?w=1

### CODE REVIEW

**Comment by ljharb** [comment]
LGTM. this isn't even a breaking change, which is nice (just a minor)

**Comment by ljharb** [comment]
This will now need a rebase due to #9

**Comment by JakeChampion** [comment]
@ljharb Rebased :-)

File: array-of.js [hunk 1]
```diff
@@ -1,56 +1,54 @@
...
+	if (defineProperty) {
+		defineProperty(Array, 'of', {
+			'value': of,
+			'configurable': true,
+			'writable': true
+		});
+	} else {
+		Array.of = of;
+	}
+}());
```
Comment (ljharb) [inline array-of.js]: can you add a trailing newline?

**Comment by ljharb** [comment]
LGTM after the last nit :-)

**Comment by JakeChampion** [comment]
@ljharb Done.

**Pull request merged**